src/main/java/taa/command/DeleteStudentCommand.java

```diff
@@ -51,6 +51,10 @@ public void execute(ModuleList moduleList, Ui ui) throws TaaException {
         int studentIndex = Integer.parseInt(studentIndexInput) - 1;
 
         StudentList studentList = module.getStudentList();
+        if (studentIndex < 0 || studentIndex >= studentList.getSize()) {
```

Comment:
This check is redundant as `studentList.deleteStudent(studentIndex);` at line 58 will return null if it fails to delete due to invalid index.